tmt/utils.py

```diff
@@ -6183,3 +6183,29 @@ def __exit__(self, *args: Any) -> None:
     @property
     def duration(self) -> datetime.timedelta:
         return self.end_time - self.start_time
+
+
+def retry(
+        function_to_retry: Callable[..., Any],
+        retry_count: int,
+        delay: int,
+        error_message: str,
+        *args: Optional[Any],
+        **kwargs: Optional[Any]
+        ) -> Any:
+    """ Retry functionality to be used elsewhere in the code.
+
+    :param function_to_retry: function to be called with any amount
+    of arguments of Any type, returning result of Any type
+    :param retry_count: number of tries to call the function
+    :param delay: amount of seconds to wait before a new try
+    :param error_message: message explaining retry reason
+    """
+
+    for _ in range(retry_count):
+        try:
+            function_to_retry(*args, **kwargs)
+            break
+        except Exception:
+            print(f"{error_message}")
+            time.sleep(delay)
```

Comment:
And what happens when we run out of retries? In this state, the `retry` function would return like nothing happened :)

Check out `tmt.utils.wait()` - we have a custom exception raised when waiting takes too much time, I suppose we should have one to raise when `retry` runs out of retries. That would be its final step.